src/main/java/forestry/core/genetics/SaveFormat.java

```diff
@@ -0,0 +1,218 @@
+package forestry.core.genetics;
+
+import javax.annotation.Nullable;
+
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+
+import forestry.api.genetics.IAllele;
+import forestry.api.genetics.IChromosome;
+import forestry.api.genetics.IChromosomeType;
+import forestry.api.genetics.ISpeciesRoot;
+import forestry.core.utils.SimpleByteBuf;
+
+public enum SaveFormat {
+	//Used before forge fires the FMLLoadCompleteEvent.
```

Comment:
Can you explain why this is needed? What is different before `FMLLoadCompleteEvent`?

Reply:
The forge registry uses the ids that are saved in the `level.dat` file. All genetic items that we have created before the world was loaded would break, if we would use the `BINARY` format and the player would remove a mod that registers alleles. Because the items would use the ids from the a version of the registry that has never seen the old alleles.

Reply:
That makes sense. I thought about the edge cases here and this is a nice solution.